src/test/groovy/com/palantir/gradle/versions/VersionPropsIdeaPluginIntegrationSpec.groovy

```diff
@@ -0,0 +1,84 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions
+
+import com.fasterxml.jackson.dataformat.xml.XmlMapper
+import com.fasterxml.jackson.datatype.guava.GuavaModule
+import nebula.test.IntegrationSpec
+import spock.util.environment.RestoreSystemProperties
+
+import java.util.stream.Collectors
+
+class VersionPropsIdeaPluginIntegrationSpec extends IntegrationSpec {
+
+    def setup() {
+        //language=gradle
+        buildFile << """
+            repositories {
+                maven {
+                    url 'https://test'
+                }
+                maven {
+                    url 'https://demo/'
+                }
+                mavenCentral() { metadataSources { mavenPom(); ignoreGradleMetadataRedirection() } }
+            }
+
+            apply plugin: 'com.palantir.version-props-idea'
+            apply plugin: 'idea'
+        """.stripIndent(true)
+
+        def ideaDir = new File(projectDir, '.idea')
+        ideaDir.mkdirs()
+
+        System.setProperty('idea.active', 'true')
```

Comment:
I think it's best not to test like this using `System.setProperty` as it will set the property on all further tests. `@RestoreSystemProperties` is a bit dangerous as you have to remember to add it to each test. Instead, you can set the system property as part of the gradle invocation:

```
runTasksSuccessfully('-Didea.active=true')
```